annotated.go

```diff
@@ -126,6 +130,66 @@ func newAnnotations(fType reflect.Type) annotations {
 	}
 }
 
+// field used for embedding fx.Out type in generated struct.
+var _outAnnotationField = reflect.StructField{
+	Name:      "Out",
+	Type:      reflect.TypeOf(Out{}),
+	Anonymous: true,
+}
+
+// This generates the annotated Out struct when function is annotated
+// with fx.As and/or fx.ResultTags options.
+func (ann *annotations) genAnnotatedOutStruct() error {
+	if !ann.annotatedOut && !ann.annotatedAs {
+		return nil
+	}
+	fType := ann.fType
+	offsets := make([]int, fType.NumOut())
```

Comment:
A comment here would be helpful:

```suggestion
	// offsets[i] is the index of this field in annotatedResult.
	// This will always be >0 when valid.
	offsets := make([]int, fType.NumOut())
```